### DISCUSSION TITLE
Question about IDE configuration for generated GEM sources during development

### DISCUSSION THREAD

**Discussion by fotgu** [mapstruct/mapstruct#3764] [Q&A] [answered]
Hi MapStruct team,

I'm currently working with MapStruct's codebase and noticed that during development, extra Intelij IDEA configuration is needed for recognizing GEM-generated sources (like MapperGem) in the processor/target/generated-sources/annotations directory. Specifically, these generated files need to be manually marked as "Generated Sources Root" in the IDE to be properly recognized.

I might be missing something, but I was wondering if there are any existing configurations or recommended practices for handling these generated source directories automatically in development environments?

If there isn't an existing solution, I found that adding the build-helper-maven-plugin to processor/pom.xml could automatically mark the generated sources directory for IDEs. Here's the specific change I'm thinking about:
```xml
<build>
    <plugins>
        <plugin>
            <groupId>org.codehaus.mojo</groupId>
            <artifactId>build-helper-maven-plugin</artifactId>
            <executions>
                <execution>
                    <id>add-source</id>
                    <phase>generate-sources</phase>
                    <goals>
                        <goal>add-source</goal>
                    </goals>
                    <configuration>
                        <sources>
                            <source>${project.build.directory}/generated-sources/annotations</source>
                        </sources>
                    </configuration>
                </execution>
            </executions>
        </plugin>
    </plugins>
</build>
```
I'm curious about your thoughts on this approach and whether there might be specific reasons for the current setup.

Thank you for your time and for maintaining this excellent project.

**Comment by filiphr** [comment]
Which IntelliJ version are you using @fotgu? For me IntelliJ is automatically making the `processor/target/generated-sources/annotations ` folder as generated source and everything works as expected.

**Reply by fotgu** [reply]
> Which IntelliJ version are you using @fotgu? For me IntelliJ is automatically making the `processor/target/generated-sources/annotations ` folder as generated source and everything works as expected.

I'm using IntelliJ IDEA 2024.3 (Ultimate Edition), the detailed version is `Build #IU-243.21565.193, built on November 13, 2024`.
I just tried reimporting the mapstruct codebase into IDEA, but it still needs manual configuration. @filiphr

**Reply by filiphr** [reply, marked as answer]
I just tried it with a fresh clone and a new IntelliJ project. I did the following steps:

* New project from existing sources
* Use maven project
* Wait for Index
* Build Project through Build -> Build Project

At this point the build failed and the `processor/target/generated-sources/annotations`. What I did next is the Maven "Reload All Maven Projects"

![image](https://github.com/user-attachments/assets/31f61614-a327-43dd-8024-156d537c013a)

After doing the that the folder became part of my sources.
